server/src/main/java/io/seata/server/storage/redis/lock/RedisLocker.java

```diff
@@ -0,0 +1,256 @@
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.seata.server.storage.redis.lock;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.alibaba.fastjson.JSON;
+
+import io.seata.common.util.CollectionUtils;
+import io.seata.common.util.LambdaUtils;
+import io.seata.common.util.StringUtils;
+import io.seata.config.ConfigurationFactory;
+import io.seata.core.constants.ConfigurationKeys;
+import io.seata.core.lock.AbstractLocker;
+import io.seata.core.lock.RowLock;
+import io.seata.core.store.LockDO;
+import io.seata.server.storage.redis.JedisPooledFactory;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.Pipeline;
+
+/**
+ * @author funkye
+ */
+public class RedisLocker extends AbstractLocker {
+
+    private static final Integer DEFAULT_SECONDS = 30;
+
+    private static final Integer DEFAULT_QUERY_LIMIT = 100;
+
+    private static final String DEFAULT_REDIS_SEATA_LOCK_PREFIX = "SEATA_LOCK_";
+
+    private static final String DEFAULT_REDIS_SEATA_LOCK_XID_PREFIX = "SEATA_LOCK_XID_";
+
+    /**
+     * The query limit.
+     */
+    private int logQueryLimit;
+
+    /**
+     * Instantiates a new Redis locker.
+     *
+     */
+    public RedisLocker() {
+        logQueryLimit =
+            ConfigurationFactory.getInstance().getInt(ConfigurationKeys.STORE_REDIS_QUERY_LIMIT, DEFAULT_QUERY_LIMIT);
+    }
+
+    @Override
+    public boolean acquireLock(List<RowLock> rowLocks) {
+        if (CollectionUtils.isEmpty(rowLocks)) {
+            // no lock
+            return true;
+        }
+        List<String> successList = new ArrayList<>();
+        long status = 1;
+        try (Jedis jedis = JedisPooledFactory.getJedisInstance()) {
+            List<LockDO> locks = convertToLockDO(rowLocks);
+            if (locks.size() > 1) {
+                locks =
+                    locks.stream().filter(LambdaUtils.distinctByKey(LockDO::getRowKey)).collect(Collectors.toList());
+            }
+            Pipeline pipeline = jedis.pipelined();
+            for (LockDO lock : locks) {
+                String key = getLockKey(lock.getRowKey());
+                pipeline.setnx(key, JSON.toJSONString(lock));
+            }
+            List<Object> results = pipeline.syncAndReturnAll();
+            for (int i = 0; i < results.size(); i++) {
+                Long result = (long)results.get(i);
+                String key = getLockKey(locks.get(i).getRowKey());
+                if (result != 1) {
+                    status = result;
```

Comment:
break?

Reply:
> break?

meed to record the successfully inserted key

Reply:
my fault